### PR TITLE
doc(README): add examples for GPU sharing and update-examples

### PR DESCRIPTION
**Pull request by xiaoyao**
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:
Add usage examples for GPU sharing in readme for newers

**Which issue(s) this PR fixes**:
Help avoiding #661 #590

**Special notes for your reviewer**:
- See more info in: https://github.com/Project-HAMi/HAMi/issues/590#issuecomment-2510661848
- Unify the unit of `nvidia.com/gpumem` in examples to M.

**Does this PR introduce a user-facing change?**: no

### CODE REVIEW

File: example.yaml [hunk 1]
```diff
@@ -35,7 +35,7 @@ spec:
           resources:
             limits:
               nvidia.com/gpu: 1 # requesting 1 vGPUs
```
Comment (wawa0210) [inline example.yaml]: `nvidia.com/gpu` Declare how many physical GPUs the pod needs,

`nvidia.com/gpumem`, `nvidia.com/gpucores`  Identifies how much GPU memory and GPU core each physical GPU allocates to the pod

File: README_cn.md [hunk 2]
```diff
@@ -43,24 +43,45 @@ HAMi 是[Cloud Native Computing Foundation](https://cncf.io/)(CNCF)基金会的s
...
+      resources:
+        limits:
+          nvidia.com/gpu: 2 # requesting 2 vGPUs
+          nvidia.com/gpumem: 10000 # Each vGPU containers 10G device memory
+          # nvida.com/gpucores: 50 # Utilization limit of this vGPU is set to 50% of total GPU utilization 
+...
+# pod-B of UserB
+      resources:
+        limits:
+          nvidia.com/gpu: 2 # requesting 2 vGPUs
```
Comment (wawa0210) [inline README_cn.md]: `nvidia.com/gpu` Declare how many physical GPUs the pod needs,

`nvidia.com/gpumem`, `nvidia.com/gpucores` Identifies how much GPU memory and GPU core each physical GPU allocates to the pod

File: README.md [hunk 3]
```diff
@@ -35,15 +35,37 @@ HAMi is a sandbox and [landscape](https://landscape.cncf.io/?item=orchestration-
...
 
 ### Device sharing
-
+HAMi supports:
 - Allows partial device allocation by specifying device memory.
-- Imposes a hard limit on streaming multiprocessors.
 - Permits partial device allocation by specifying device core usage.
 - Requires zero changes to existing programs.
 
 <img src="./imgs/example.png" width = "500" />
```
Comment (wawa0210) [inline README.md]: I think it is necessary to focus on the meaning of the fields `nvidia.com/gpu`,`nvidia.com/gpumem`,`nvida.com/gpucores`. The detailed examples have special yaml and do not need to be presented here.

File: README.md [hunk 4]
```diff
@@ -51,14 +51,18 @@ A task with the following resources:
 ```
       resources:
         limits:
-          nvidia.com/gpu: 1 # requesting 1 vGPU
-          nvidia.com/gpumem: 3000 # Each vGPU contains 3000m device memory
+          nvidia.com/gpu: 1 # Declare how many physical GPUs the pod needs
```
Comment (wawa0210) [inline README.md]: In common practice, inline comments start with a lowercase letter, while standalone comments begin with an uppercase letter.

such as: https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer
https://kubernetes.io/docs/concepts/services-networking/service/#services-without-selectors

File: README_cn.md [hunk 5]
```diff
@@ -55,6 +55,10 @@ HAMi支持设备资源的硬隔离
 
 ![img](./imgs/hard_limit.jpg)
 
+> 注意：
+1. **安装HAMi后，节点上注册的 `nvidia.com/gpu` 值默认为“vGPU数量”**
```
Comment (wawa0210) [inline README_cn.md]: “vGPU数量” 

The reason for using quotation marks?

File: docs/develop/tasklist.md [hunk 6]
```diff
@@ -113,6 +113,6 @@ spec:
       command:["bash"，"-c"，"sleep 86400"]
       resources:
         limits:
-          nvidia.com/gpu: 2 # requesting 2 VGPUs
+          nvidia.com/gpu: 2 # Declare how many physical GPUs the pod needs
```
Comment (wawa0210) [inline docs/develop/tasklist.md]: Same as above.

File: examples/nvidia/use_sharing_card.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    - name: ubuntu-container
+      image: ubuntu:18.04
+      command: ["bash", "-c", "sleep 86400"]
+      resources:
+        limits:
+          nvidia.com/gpu: 1 # Declare how many physical GPUs the pod needs
+          nvidia.com/gpumem-percentage: 60 # Identifies 60% GPU memory each physical GPU allocates to the pod （Optional,Integer)
+          nvidia.com/gpucores: 40 # Identifies 40% GPU GPU core each physical GPU allocates to the pod （Optional,Integer)
+
+# gpu-pod1 and gpu-pod2 could share the same GPU
```
Comment (wawa0210) [inline examples/nvidia/use_sharing_card.yaml]: “It is recommended to place this at the very beginning of the document.”

File: README.md (hunk 4 above)
Comment (xiaoyao) [reply README.md]: Got! The little diference makes perfect sense. I've learned.

File: README_cn.md (hunk 5 above)
Comment (xiaoyao) [reply README_cn.md]: Originally intended to emphasize. Already removed the quotation marks.

**Review by wawa0210 — APPROVED** [review]

**Review by wawa0210 — APPROVED** [review]

**Comment by wawa0210** [comment]
/lgtm

thanks your first contribution and looking forward to more participation

**Pull request merged**